hazelcast/src/main/java/com/hazelcast/sql/impl/type/converter/AbstractStringConverter.java

```diff
@@ -172,6 +174,40 @@ public final Object convertToSelf(Converter valConverter, Object val) {
 
     protected abstract String cast(Object val);
 
+    /**
+     * Parse non-standard dates with possible leading zeroes.
+     * Before method tried to parse such a date, potential date candidate
+     */
+    private LocalDate parseDateWithPossibleLeadingZeroes(Object val) {
+        if (!(val instanceof String)) {
+            throw new DateTimeParseException("Can't parse date from non-String source", "", 0);
+        }
+
+        // This pattern checks both cases with and without leading zeroes in date.
+        Pattern pattern = Pattern.compile("\\d{4}-([1-9]|[0-1][0-2])-([1-9]|[0-2][0-9]|3[01])$");
```

Comment:
Probably not an actual issue, but IDEA's regex evaluator seems to fail on this date: 
`2020-09-11` input (leading zero on the month that is >= March). Tests work correctly with this case though: 
```
assertEquals(LocalDate.parse("2020-09-11"), c.asDate("2020-09-11"));
```
Changing the regex to include the possible leading zero case fixes the issue for IDEA's evaluator: 
```
\d{4}-([0]*[1-9]|[0-1][0-2])-([1-9]|[0-2][0-9]|3[01])$
```
(there's probably a more elegant way to include this possible zero predicate though)

Not sure what's up with that, I'm probably missing some context as to why the original Regex is supposed to work as is, but thought its worth mentioning nonetheless.

Reply:
Think I figured it out, this is the case that would need this change: 
```
assertEquals(LocalDate.parse("2020-09-11"), c.asDate("2020-09-1"));
```
Not sure if we need to support that also.